test/python/pulse/test_schedule.py

```diff
@@ -307,6 +308,119 @@ def test_flatten(self):
 
         self.assertEqual(flat_sched.instructions, sched.instructions)
 
+    def test_filter_channels(self):
+        """Test filtering over channels."""
+        device = self.two_qubit_device
+        lp0 = self.linear(duration=3, slope=0.2, intercept=0.1)
+        acquire = Acquire(5)
+        sched = Schedule(name='fake_experiment')
+        sched = sched.insert(0, lp0(device.q[0].drive))
+        sched = sched.insert(10, lp0(device.q[1].drive))
+        sched = sched.insert(30, FrameChange(phase=-1.57)(device.q[0].drive))
+        sched = sched.insert(60, acquire(device.q, device.mem))
```

Comment:
@taalexander The acquire instruction that gets added here doesn't contain an `AcquireChannel(0)` like I'd expected. Do you know what's going on there?

Reply:
The acquire instruction knows how that each `Qubit` can only have one acquire. As `device.q` is a list of `Qubit`s so beind the scenes it does `[q.acquire for q in device.q]`. I am not sure how I feel about this, on one hand, it easy shorthand, but on the other hand, it is magic.